src/seedu/addressbook/data/person/Address.java

```diff
@@ -8,52 +8,153 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
-
-    public final String value;
-    private boolean isPrivate;
-
-    /**
-     * Validates given address.
-     *
-     * @throws IllegalValueException if given address string is invalid.
-     */
-    public Address(String address, boolean isPrivate) throws IllegalValueException {
-        String trimmedAddress = address.trim();
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
-        }
-        this.value = trimmedAddress;
-    }
-
-    /**
-     * Returns true if a given string is a valid person email.
-     */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
-    }
-
-    @Override
-    public String toString() {
-        return value;
-    }
-
-    @Override
-    public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
-    }
-
-    @Override
-    public int hashCode() {
-        return value.hashCode();
-    }
-
-    public boolean isPrivate() {
-        return isPrivate;
-    }
+	public static final String EXAMPLE = "123, some street";
+	public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses must be in "
+			+ "the a/BLOCK, STREET, UNIT, POSTAL_CODE format";
+	public static final String ADDRESS_VALIDATION_REGEX = ", ";
+	public static final Integer BLOCK_INDEX_IN_ADDRESS = 0;
+	public static final Integer STREET_INDEX_IN_ADDRESS = 1;
+	public static final Integer UNIT_INDEX_IN_ADDRESS = 2;
+	public static final Integer POSTALCODE_INDEX_IN_ADDRESS = 3;
+
+	private static final int ADDRESS_COMPONENTS = 4;
+
+	public final Block block;
+	public final Street street;
+	public final Unit unit;
+	public final PostalCode postalCode;
+
+	public final String value;
```

Comment:
Why are you still maintaining the 'value' variable? It is duplicate data. Use the new classes instead.